telegram/ext/jobqueue.py

```diff
@@ -349,34 +346,24 @@ def run_monthly(
         job = Job(callback, context, name, self)
 
         # if day == -1 the run on last day of the month
-        if day == -1:
-            j = self.scheduler.add_job(
-                callback,
-                trigger='cron',
-                args=self._build_args(job),
-                name=name,
-                day='last',
-                hour=when.hour,
-                minute=when.minute,
-                second=when.second,
-                timezone=when.tzinfo or self.scheduler.timezone,
-                **job_kwargs,
-            )
         # else run on the day as provided
+        if day == -1:
+            _day = 'last'
         else:
-            j = self.scheduler.add_job(
-                callback,
-                trigger='cron',
-                args=self._build_args(job),
-                name=name,
-                day=day,
-                hour=when.hour,
-                minute=when.minute,
-                second=when.second,
-                timezone=when.tzinfo or self.scheduler.timezone,
-                **job_kwargs,
-            )
+            _day = day
 
+        j = self.scheduler.add_job(
+            callback,
+            trigger='cron',
+            args=self._build_args(job),
+            name=name,
+            day=_day,
```

Comment:
Better :) but we can still shorten it by replacing this line by
```suggestion
            day='last' is day == -1 else day,
```
😉 that shuld also fix the failing pre-commit test